clang/test/SemaCXX/lambda-expressions.cpp

```diff
@@ -718,3 +718,8 @@ void foo() {
 void GH48527() {
   auto a = []()__attribute__((b(({ return 0; })))){}; // expected-warning {{unknown attribute 'b' ignored}}
 }
+
+void GH67492() {
+  constexpr auto test = 42;
+  auto lambda = (test, []() noexcept(true) {});
```

Comment:
Does this really attempt a capture?  You have `noexcept(true)` here?

Reply:
Yes, that's exactly the bug. It attempt a capture because we are in the lambda (in term of declaration context/scope),
but we were in the evaluation context of the enclosing expression.

`ActOnFinishFullExpression` was called (on `true`), we were thinking that `test` was in that evaluation context, and we were attempting to mark it odr used... from within the declaration context of the lambda, it's beautiful

https://godbolt.org/z/Waosza81K

Reply:
Which one?

Reply:
I can't remember... you asked my advise on one the other day where a requires clause inside of a thing inside of a lambda (really long repro) was trying to capture and I suggested something similar to this one (checking the contexts were added correctly).